app/src/test/java/org/greenstand/android/TreeTracker/signup/SignupViewModelTest.kt

```diff
@@ -0,0 +1,194 @@
+        Truth.assertThat(signupViewModel.state.getOrAwaitValueTest().phone).isNull()
+        Truth.assertThat(signupViewModel.state.getOrAwaitValueTest().email).containsMatch("@")
+
+
+    }
+
+    @Test
+    fun `update email contains invalid credentials, returns Error`() = runBlocking{
+        signupViewModel.updateEmail("somerandom!!gmsail.com")
+        Truth.assertThat(signupViewModel.state.getOrAwaitValueTest().email).doesNotContainMatch("@")
+    }
+
+    // Phone
+
+    @Test
+    fun `update phone contains valid credentials, returns success`() = runBlocking{
+        signupViewModel.updatePhone("071321998893")
+        Truth.assertThat(signupViewModel.state.getOrAwaitValueTest().isCredentialValid).isTrue()
+        Truth.assertThat(signupViewModel.state.getOrAwaitValueTest().email).isNull()
+
+    }
+
+    @Test
+    fun `update phone contains invalid credentials, returns Error`() = runBlocking{
+        signupViewModel.updatePhone("random1919")
+        Truth.assertThat(signupViewModel.state.getOrAwaitValueTest().isCredentialValid).isFalse()
+    }
+
+    // Selfie Tutorial Dialogs
+    @Test
+    fun `update selfie tutorial dialog state = false, returns false`() = runBlocking{
+        signupViewModel.updateSelfieTutorialDialog(false)
+        Truth.assertThat(signupViewModel.state.getOrAwaitValueTest().showSelfieTutorial).isFalse()
+    }
+
+    @Test
+    fun `update selfie tutorial dialog state = true, returns true`() = runBlocking{
+        signupViewModel.updateSelfieTutorialDialog(true)
+        Truth.assertThat(signupViewModel.state.getOrAwaitValueTest().showSelfieTutorial).isTrue()
+    }
+
+    // Go to Credentials Entry
+    @Test
+    fun `Go to credentials Entry, returns true`() = runBlocking{
+        signupViewModel.goToCredentialEntry()
+        Truth.assertThat(signupViewModel.state.getOrAwaitValueTest().isCredentialView).isTrue()
+        Truth.assertThat(signupViewModel.state.getOrAwaitValueTest().firstName).isNull()
+        Truth.assertThat(signupViewModel.state.getOrAwaitValueTest().lastName).isNull()
+        Truth.assertThat(signupViewModel.state.getOrAwaitValueTest().canGoToNextScreen).isTrue()
+    }
+
+    // Enables Autofocus
+    @Test
+    fun `enable autofocus, returns true`() = runBlocking{
+        signupViewModel.enableAutofocus()
+        Truth.assertThat(signupViewModel.state.getOrAwaitValueTest().autofocusTextEnabled).isTrue()
+    }
+
+    // Close privacy policy dialog
+    @Test
+    fun `close privacy policy dialog, returns true`() = runBlocking{
+        signupViewModel.closePrivacyPolicyDialog()
+        Truth.assertThat(signupViewModel.state.getOrAwaitValueTest().showPrivacyDialog).isFalse()
+    }
+
+    // Close Existing User Dialog
+    @Test
+    fun `close existing user dialog, returns true`() = runBlocking{
+        signupViewModel.closeExistingUserDialog()
+        Truth.assertThat(signupViewModel.state.getOrAwaitValueTest().existingUser).isNull()
+    }
+
+}
+

```

Comment:
Please remove all this extra spacing at the bottom

Reply:
sure